vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -469,3 +611,51 @@ pub(crate) fn admin_cmd(opcode: spec::AdminOpcode) -> spec::Command {
         ..FromZeroes::new_zeroed()
     }
 }
+
+#[repr(C)]
+#[derive(Protobuf, Clone, Debug, AsBytes, FromBytes, FromZeroes)]
+#[mesh(package = "underhill")]
+pub struct PendingCommandSavedState {
+    #[mesh(1)]
+    pub command: [u8; 64],
+    #[mesh(2)]
+    pub cid: u16,
+}
+
+impl From<&[u8]> for PendingCommandSavedState {
+    fn from(value: &[u8]) -> Self {
+        let mut command: [u8; 64] = [0; 64];
+        command.copy_from_slice(value);
+        let cid = ((command[0] as u16) << 8) | command[1] as u16;
+        Self { command, cid }
+    }
+}
+
+#[derive(Protobuf, Clone, Debug)]
+#[mesh(package = "underhill")]
+pub struct QueuePairSavedState {
+    #[mesh(1)]
+    /// Which CPU handles requests.
+    pub cpu: u32,
+    #[mesh(2)]
+    /// Interrupt vector (MSI-X)
+    pub msix: u32,
+    #[mesh(3)]
+    pub max_cids: usize,
+    #[mesh(4)]
+    pub sq_state: SubmissionQueueSavedState,
+    #[mesh(5)]
+    pub cq_state: CompletionQueueSavedState,
+    #[mesh(6)]
+    pub sq_addr: u64,
```

Comment:
These and some other fields seem redundant with stuff that's in `SubmissionQueueSavedState`/`CompletionQueueSavedState`.

Reply:
Will review redundancy and eliminate all saved VA addresses